tests/unit/neptune/new/client/test_project.py

```diff
@@ -131,3 +138,49 @@ def test_prepare_nql_query():
         trashed=None,
     )
     assert len(query.items) == 0
+
+
+def test_parse_dates():
+    def entries_generator():
+        yield LeaderboardEntry(
+            id="test",
+            attributes=[
+                AttributeWithProperties(
+                    "attr1",
+                    AttributeType.DATETIME,
+                    {"value": datetime(2024, 2, 5, 20, 37, 40, 915000).strftime(DATE_FORMAT)},
+                ),
+                AttributeWithProperties(
+                    "attr2",
+                    AttributeType.DATETIME,
+                    {"value": datetime(2024, 2, 5, 20, 37, 40, 915000).strftime(DATE_FORMAT)},
+                ),
+            ],
+        )
+
+    parsed = list(parse_dates(entries_generator()))
+    assert parsed[0].attributes[0].properties["value"] == datetime(2024, 2, 5, 20, 37, 40, 915000)
+    assert parsed[0].attributes[1].properties["value"] == datetime(2024, 2, 5, 20, 37, 40, 915000)
+
+
+@patch("neptune.metadata_containers.utils.warn_once")
+def test_parse_dates_wrong_format(mock_warn_once):
+    entries = [
+        LeaderboardEntry(
+            id="test",
+            attributes=[
+                AttributeWithProperties(
+                    "attr1",
+                    AttributeType.DATETIME,
+                    {"value": "07-02-2024"},  # different format than expected
+                )
+            ],
+        )
+    ]
+
+    parsed = list(parse_dates(entries))
+    assert parsed[0].attributes[0].properties["value"] == "07-02-2024"  # should be left unchanged due to ValueError
+    mock_warn_once.assert_called_once_with(
+        "Date parsing failed. The date format is incorrect. Returning in the string format.",
```

Comment:
Maybe something slightly more helpful. Like this, if I understand it correctly

```suggestion
        "Date parsing failed. The date format is incorrect. Returning as string instead of datetime.",
```

Reply:
Done